src/managers/notification-manager.ts

```diff
@@ -21,65 +29,73 @@ export interface ToastActionParams {
 class NotificationManager extends LitElement {
   @property({ attribute: false }) public hass!: HomeAssistant;
 
-  @state() private _action?: ToastActionParams;
-
-  @state() private _noCancelOnOutsideClick = false;
+  @state() private _parameters?: ShowToastParams;
 
   @query("ha-toast") private _toast!: HaToast;
 
-  public async showDialog({
-    message,
-    action,
-    duration,
-    dismissable,
-  }: ShowToastParams) {
-    let toast = this._toast;
+  public async showDialog(parameters: ShowToastParams) {
     // Can happen on initial load
-    if (!toast) {
+    if (!this._toast) {
       await this.updateComplete;
-      toast = this._toast;
     }
-    toast.setAttribute("dir", computeRTL(this.hass) ? "rtl" : "ltr");
-    this._action = action || undefined;
-    this._noCancelOnOutsideClick =
-      dismissable === undefined ? false : !dismissable;
-    toast.hide();
-    toast.show({
-      text: message,
-      duration: duration === undefined ? 3000 : duration,
-    });
+    this._toast.close("dismiss");
+    this._parameters = parameters;
+    if (!this._parameters || this._parameters.duration === 0) {
+      return;
+    }
+
+    if (
+      this._parameters.duration !== undefined &&
+      this._parameters.duration > 0 &&
+      this._parameters.duration <= 4000
+    ) {
+      this._parameters.duration = 4000;
+    }
+
+    this._toast.labelText = this._parameters.message;
+    if (this._parameters.duration) {
+      this._toast.timeoutMs = this._parameters.duration;
+    }
```

Comment:
Shouldn't we just set this in `render`?

Reply:
I would keep a similar design as we use for `dialogs`, use the `open` attribute on the `snackbar` and render `nothing` when the snackbar is closed. Use the `MDCSnackbar:closed` event for that.